test/mainnet-fork/CCTPCalls.t.sol

```diff
@@ -255,6 +246,47 @@ contract USDCToCCTPIntegrationTests is BaseChainUSDCToCCTPTestBase {
         assertEq(usdcBase.totalSupply(),                         USDC_BASE_SUPPLY + 1e6);
     }
 
+    function test_transferUSDCToCCTP_sourceToDestination_bigTransfer() external {
+        deal(address(usdc), address(almProxy), 1_900_000e6);
+
+        assertEq(usdc.balanceOf(address(almProxy)),          1_900_000e6);
+        assertEq(usdc.balanceOf(address(mainnetController)), 0);
+        assertEq(usdc.totalSupply(),                         USDC_SUPPLY);
+
+        assertEq(usds.allowance(address(almProxy), CCTP_MESSENGER),  0);
+
+        // Will split into two separate transactions at max 1m each
+        _expectEthereumCCTPEmit(
+            94773,
+            1_000_000e6
+        );
+        _expectEthereumCCTPEmit(
+            94774,
+            900_000e6
+        );
+
+        vm.prank(relayer);
+        mainnetController.transferUSDCToCCTP(1_900_000e6, CCTPForwarder.DOMAIN_ID_CIRCLE_BASE);
+
+        assertEq(usdc.balanceOf(address(almProxy)),          0);
+        assertEq(usdc.balanceOf(address(mainnetController)), 0);
+        assertEq(usdc.totalSupply(),                         USDC_SUPPLY - 1_900_000e6);
+
+        assertEq(usds.allowance(address(almProxy), CCTP_MESSENGER),  0);
+
+        destination.selectFork();
+
+        assertEq(usdcBase.balanceOf(address(foreignAlmProxy)),   0);
+        assertEq(usdcBase.balanceOf(address(foreignController)), 0);
+        assertEq(usdcBase.totalSupply(),                         USDC_BASE_SUPPLY);
+
+        bridge.relayMessagesToDestination(true);
+
+        assertEq(usdcBase.balanceOf(address(foreignAlmProxy)),   1_900_000e6);
+        assertEq(usdcBase.balanceOf(address(foreignController)), 0);
+        assertEq(usdcBase.totalSupply(),                         USDC_BASE_SUPPLY + 1_900_000e6);
+    }
```

Comment:
Update to do multiple iterations of while loop:


```suggestion
    function test_transferUSDCToCCTP_sourceToDestination_bigTransfer() external {
        deal(address(usdc), address(almProxy), 2_900_000e6);

        assertEq(usdc.balanceOf(address(almProxy)),          2_900_000e6);
        assertEq(usdc.balanceOf(address(mainnetController)), 0);
        assertEq(usdc.totalSupply(),                         USDC_SUPPLY);

        assertEq(usds.allowance(address(almProxy), CCTP_MESSENGER),  0);

        // Will split into two separate transactions at max 1m each
        _expectEthereumCCTPEmit(
            94773,
            1_000_000e6
        );
        _expectEthereumCCTPEmit(
            94774,
            1_000_000e6
        );
        _expectEthereumCCTPEmit(
            94775,
            900_000e6
        );

        vm.prank(relayer);
        mainnetController.transferUSDCToCCTP(2_900_000e6, CCTPForwarder.DOMAIN_ID_CIRCLE_BASE);

        assertEq(usdc.balanceOf(address(almProxy)),          0);
        assertEq(usdc.balanceOf(address(mainnetController)), 0);
        assertEq(usdc.totalSupply(),                         USDC_SUPPLY - 2_900_000e6);

        assertEq(usds.allowance(address(almProxy), CCTP_MESSENGER),  0);

        destination.selectFork();

        assertEq(usdcBase.balanceOf(address(foreignAlmProxy)),   0);
        assertEq(usdcBase.balanceOf(address(foreignController)), 0);
        assertEq(usdcBase.totalSupply(),                         USDC_BASE_SUPPLY);

        bridge.relayMessagesToDestination(true);

        assertEq(usdcBase.balanceOf(address(foreignAlmProxy)),   2_900_000e6);
        assertEq(usdcBase.balanceOf(address(foreignController)), 0);
        assertEq(usdcBase.totalSupply(),                         USDC_BASE_SUPPLY + 2_900_000e6);
    }
```

Reply:
updated